src/modules/findProvinceFromCoordinate/index.ts

```diff
@@ -0,0 +1,74 @@
+import { GeoJSONData } from "./IRGeoJSON";
+
+interface Point {
+	longitude: number;
+	latitude: number;
+}
+
+interface GeoJSONFeature {
+	properties: Record<string, string | undefined>;
+	geometry: {
+		type: string;
+		coordinates: object; // This represents MultiPolygon coordinates
+	};
+}
+
+interface Province {
+	properties: {
+		'name:fa': string;
+		'name:en': string;
+	};
+	geometry: object; 
+}
+
+const provinces = GeoJSONData;
+
+function pointInPolygon(polygon: number[][], point: Point): boolean {
+	let odd = false;
+	for (let i = 0, j = polygon.length - 1; i < polygon.length; i++) {
+		if (((polygon[i][1] > point.latitude) !== (polygon[j][1] > point.latitude))
+			&& (point.longitude < ((polygon[j][0] - polygon[i][0]) * (point.latitude - polygon[i][1]) / (polygon[j][1] - polygon[i][1]) + polygon[i][0]))) {
```

Comment:
using the Ray casting algorithm is good idea here but this if statement is so cluttered and hard to read 

you could define some local variables to hold values instead of using them directly like `polygon[i][1]`

```js
const xi = polygon[i][0];
const yi = polygon[i][1];
const xj = polygon[j][0]; 
const yj = polygon[j][1];


const {  longitude: x, latitude: y } = point

// so much simpler to read 
((yi > y) !== (yj > y)) && (x < (xj - xi) * (y - yi) / (yj - yi) + xi);
```